lib/matplotlib/quiver.py

```diff
@@ -57,110 +52,124 @@
 scale down all the head parameters. You will probably do best to leave
 minshaft alone.
 
+**Arrow outline**
+
 *linewidths* and *edgecolors* can be used to customize the arrow
 outlines.
 
 Parameters
 ----------
-X : 1D or 2D array, sequence, optional
-    The x coordinates of the arrow locations
-Y : 1D or 2D array, sequence, optional
-    The y coordinates of the arrow locations
-U : 1D or 2D array or masked array, sequence
-    The x components of the arrow vectors
-V : 1D or 2D array or masked array, sequence
-    The y components of the arrow vectors
-C : 1D or 2D array, sequence, optional
-    The arrow colors
-units : [ 'width' | 'height' | 'dots' | 'inches' | 'x' | 'y' | 'xy' ]
+X, Y : 1D or 2D array-like, optional
+    The x and y coordinates of the arrow locations.
+    
+    If not given, they will be generated as a uniform integer meshgrid based
+    on the dimensions of *U* and *V*.
+    
+    If *X* and *Y* are 1D but *U*, *V* are 2D, *X*, *Y* are expanded to 2D
+    using ``X, Y = np.meshgrid(X, Y)``. In this case ``len(X)`` and ``len(Y)``
+    must match the column and row dimensions of *U* and *V*.
+U, V : 1D or 2D array-like
```

Comment:
would suggest separating the descriptions of each parameter with a blank line (otherwise it looks a bit strange given that many individual descriptions are multi-paragraph).